cpp/ql/src/semmle/code/cpp/MemberFunction.qll

```diff
@@ -298,6 +310,12 @@ class CopyConstructor extends Constructor {
     getDeclaringType() instanceof TemplateClass and
     getNumberOfParameters() > 1
   }
+
+  override predicate hasDataFlow(FunctionInput input, FunctionOutput output) {
+    // data flow from the first constructor argument to the returned object
+    input.isParameter(0) and
+    output.isReturnValue()
```

Comment:
Somehow this works, at least for AST, but should it? A constructor conceptually doesn't have a return value, so I'd expect to see `output.isQualifierObject()` here (and elsewhere) instead.

Reply:
I don't think a constructor call has a qualifier object.  My dogs desperate for a walk, I'll check my facts after...

Reply:
Flow to the constructor qualifier doesn't work.  It plainly should, since the QLDoc for `isQualifierObject` says "the output value pointed to by the `this` pointer".  However it seems there is a `TODO` in `DataFlowUtil.qll`'s `modelFlow` to implement flow to qualifiers.

For now I'll leave it flowing to the return value, since that works in practice, with a `TODO` comment so we know to update it in future.